packages/periphery/contracts/oracles/CurveOracle.sol

```diff
@@ -92,10 +110,17 @@ contract CurveEMAPriceOracle is IPriceOracle, Ownable2Step {
     OracleParams storage poolParams = getParams[quoteToken][baseToken];
     if (poolParams.pool == address(0)) revert ZeroAddress();
 
-    uint256 price = ICurve(poolParams.pool).price_oracle();
+    uint256 price;
+    if (poolParams.priceOracleMethodHaveArg) {
```

Comment:
I'd move this if-else clause to separate `_getPriceRaw` function and call it here and in `addPool` method instead of the `_getPriceX96` one